### PR TITLE
fix(healthcheck): Update Hubble Readiness & Liveliness probes

### PR DESCRIPTION
**Pull request by mereta**
# Description
Updating Readiness probe to use hubble status command
Adding a liveliness probe to use hubble status command

## Related Issue
#1047 

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed
Checks functioning as expected. Tested with an invalid tcp address to make the gRPC connection for hubble status and containers get recycled as expected.
image

## Additional Notes
---

### CODE REVIEW

File: deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml [hunk 1]
```diff
@@ -189,11 +197,19 @@ spec:
             - -command
             - .\setkubeconfigpath.ps1; ./controller.exe --config ./retina/config.yaml --kubeconfig ./kubeconfig
           readinessProbe:
-            httpGet:
-              path: /metrics
-              port: {{ .Values.agent.container.retina.ports.containerPort }}
+            exec:
+              command:
+              - hubble
```
Comment (anubhabMajumdar) [inline deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml:202]: Windows container doesn't have `hubble` binary or support Hubble yet.

File: deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml [hunk 2]
```diff
@@ -189,11 +197,19 @@ spec:
...
+              - status
             initialDelaySeconds: 15
             periodSeconds: 10
+          livenessProbe:
+            exec:
+              command:
+              - hubble
+              - status
+            initialDelaySeconds: 30
+            periodSeconds: 30
```
Comment (matmerr) [inline deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml:212]: How long does the hubble status cmd take to respond at scale? If the command is generally quick and fits within the 30s window, perfect, but if it's frequently greater than 30s for a healthy response then restarting the pod may do more harm than good

File: deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml (hunk 1 above)
Comment (mereta) [reply deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml]: Gotcha. updated!

File: deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml (hunk 2 above)
Comment (mereta) [reply deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml]: Going to run some scale tests. TBC.

Comment (anubhabMajumdar) [reply deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml]: Ref: https://github.com/cilium/cilium/blob/c7347199af4fe426c87fc0f5572f493559259ea6/pkg/hubble/observer/local_observer.go#L210

The status should be pretty static w.r.t. scale. Should have the ring buffer size (which is fixed) as upper limit.

Comment (mereta) [reply deploy/hubble/manifests/controller/helm/retina/templates/agent/daemonset.yaml]: Also some adjustments needed to run scale tests for hubble atm

**Comment by MikeZappa87** [comment]
Do any conditions exist with Hubble status where the exit code is 0 however the retina process itself is in a bad state?

**Comment by anubhabMajumdar** [comment]
> Do any conditions exist with Hubble status where the exit code is 0 however the retina process itself is in a bad state?

Following the code of `Hubble status` [here](https://github.com/cilium/cilium/blob/c7347199af4fe426c87fc0f5572f493559259ea6/hubble/cmd/status/status.go#L77) I think there's no such scenario if `hubble status -ocompact` is not used.

**Comment by MikeZappa87** [comment]
> > Do any conditions exist with Hubble status where the exit code is 0 however the retina process itself is in a bad state?
> 
> Following the code of `Hubble status` [here](https://github.com/cilium/cilium/blob/c7347199af4fe426c87fc0f5572f493559259ea6/hubble/cmd/status/status.go#L77) I think there's no such scenario ~if `hubble status -ocompact` is not used~

If retina is deadlocked how does the current approach determine this?

**Pull request closed without merging**